src/ddmd/expressionsem.d

```diff
@@ -8771,6 +8808,100 @@ Expression binSemanticProp(BinExp e, Scope* sc)
     return null;
 }
 
+/********************************************************************************
+ * Helper function to consolidate common code for visit(BinAssignExp)
+ * visit(PowAssignExp), and visit(CatAssignExp).  It rewrites expressions
+ * of the form, for example, `s.x += 1` to `s.x(s.x() += 1)` if s.x is
+ * a getter and setter `@property`.
+ * Params:
+ *      e = the expression to rewrite
+ *      sc = the semantic scope
+ * Returns:
+ *      the rewritten expression if the procedure succeeds, an `ErrorExp` if the
+ *      and error is encountered, or `null` if `e.e1` is not a `DotIdExp`.
+ */
+Expression binAssignRewriteProp(BinAssignExp e, Scope* sc)
```

Comment:
This could be renamed to `binSemanticProp`, then you don't have to call this explicitly.